packages/models/src/resource-binding/manager.ts

```diff
@@ -0,0 +1,201 @@
+import { TypeScriptHash, LockScriptHash } from './types'
+import { ResourceBindingRegistry, ResourceBindingManagerMessage } from './interface'
+import { outPointToOutPointString } from './utils'
+import { Listener } from '@ckb-js/kuai-io'
+import type { Subscription } from 'rxjs'
+import { ChainSource } from '@ckb-js/kuai-io/lib/types'
+import { OutPointString } from '../store'
+
+export class Manager extends Actor<object, MessagePayload<ResourceBindingManagerMessage>> {
+  #registry: Map<TypeScriptHash, Map<LockScriptHash, ResourceBindingRegistry>> = new Map()
+  #registryOutpoint: Map<OutPointString, ResourceBindingRegistry> = new Map()
+  #registryReverse: Map<ActorURI, [TypeScriptHash, LockScriptHash]> = new Map()
+  #lastBlock: Block | undefined = undefined
+  #tipBlockNumber = BI.from(0)
+
+  constructor(private _listener: Listener<Header>, private _dataSource: ChainSource) {
+    super()
+  }
+
+  onListenBlock = (blockHeader: Header) => {
+    const currentBlockNumber = BI.from(blockHeader.number)
+    if (currentBlockNumber.gt(this.tipBlockNumber)) {
+      this.tipBlockNumber = currentBlockNumber
+    }
+  }
+
+  private update(pollingInterval = 1000) {
+    return setInterval(async () => {
+      if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
+        this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
+      }
+    }, pollingInterval)
+  }
+
+  private updateStore(block: Block) {
+    if (!this.#lastBlock || BI.from(block.header.number).gt(BI.from(this.#lastBlock.header.number))) {
+      const changes = this.filterCells(block)
+      for (const [store, input, data] of changes) {
+        const upgrade: { witness: HexString; cell: Cell }[] = []
+        for (const [cell, witness] of data) {
+          upgrade.push({ witness, cell })
+          const outPoint = cell.outPoint
+          if (outPoint) {
+            this.#registryOutpoint.set(outPointToOutPointString(outPoint), store)
+          }
+        }
+        this.sendMessage(store, 'update_cells', upgrade)
+
+        this.sendMessage(
+          store,
+          'remove_cell',
+          input.map((v) => v.previousOutput),
+        )
+      }
+      this.#lastBlock = block
+    }
+  }
+
+  private filterOutputs(tx: Transaction, block: Block): Map<OutPointString, [Cell, string]> {
+    const outputs = new Map<OutPointString, [Cell, string]>()
+    for (const outputIndex in tx.outputs) {
+      if (tx.hash) {
+        const outPoint = {
+          txHash: tx.hash,
+          index: BI.from(outputIndex).toHexString(),
+        }
+        outputs.set(outPointToOutPointString(outPoint), [
+          {
+            cellOutput: tx.outputs[outputIndex],
+            data: tx.outputsData[outputIndex],
+            outPoint,
+            blockHash: block.header.hash,
+            blockNumber: block.header.number,
+          },
+          tx.witnesses[outputIndex],
+        ])
```

Comment:
What if merge `witness` into the first element of the tuple?

Reply:
> What if merge `witness` into the first element of the tuple?

I use the `Cell` interface which is defined in lumos to represent the first element of the tuple , it does not contain the `witnesses` property, so I put it as the second element of the tuple.

Reply:
Got it